core/src/loader.rs

```diff
@@ -1604,38 +1609,116 @@ impl<'gc> Loader<'gc> {
     }
 
     /// Report a movie loader completion to script code.
-    fn movie_loader_complete(handle: Index, uc: &mut UpdateContext<'_, 'gc>) -> Result<(), Error> {
-        let (clip, event_handler) = match uc.load_manager.get_loader_mut(handle) {
+    fn movie_loader_complete(
+        handle: Index,
+        uc: &mut UpdateContext<'_, 'gc>,
+        dobj: Option<DisplayObject<'gc>>,
+    ) -> Result<(), Error> {
+        let (target_clip, event_handler, movie) = match uc.load_manager.get_loader_mut(handle) {
             Some(Loader::Movie {
                 target_clip,
+                movie,
                 event_handler,
                 ..
-            }) => (*target_clip, *event_handler),
+            }) => (*target_clip, *event_handler, movie.clone()),
             None => return Err(Error::Cancelled),
             _ => unreachable!(),
         };
 
+        let loader_info =
+            if let Some(MovieLoaderEventHandler::Avm2LoaderInfo(loader_info)) = event_handler {
+                Some(*loader_info.as_loader_info_object().unwrap())
+            } else {
+                None
+            };
+
+        if let Some(loader_info) = loader_info {
+            // Store the real movie into the `LoaderStream`, so that
+            // 'bytesTotal' starts returning the correct value
+            // (we previously had a fake empty SwfMovie).
+            // However, we still use `LoaderStream::NotYetLoaded`, since
+            // the actual MovieClip display object has not run its first
+            // frame yet.
+            loader_info.set_loader_stream(
+                LoaderStream::NotYetLoaded(movie.clone().unwrap(), Some(dobj.unwrap()), false),
```

Comment:
Nit: I am a bit confused by this expression: `Some(dobj.unwrap())` . `dobj` is an `Option` I believe, so I am wondering whether it could be just `dobj`. Is `unwrap()` called on it to ensure that it is present and otherwise panic?

Reply:
That's right - it should always be present here, so we should panic if it's not (instead of silently continuing).

Reply:
This actually causes a panic in AstroFlyer (#9977) while the game is loading. The error is:
`thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', core\src\loader.rs:1643:73`

Reply:
@n0samu I've updated the PR - AstroFlyer is working again.